src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -389,6 +397,10 @@ private set
             }
         }
 
+        // This property's value only relates to the top-level fields of the archive (such as the archive comment.)
+        // New entries in the archive won't change its state.
+        internal DirtyState Dirty { get; private set; }
```

Comment:
Please don't use the term `Dirty`.  I know it's used sometimes to indicate that something needs to be written but that's unnecessary jargon.  Can you instead choose a name like `ChangedState`.